web/src/message_fetch.js

```diff
@@ -326,7 +326,9 @@ export function load_messages(opts, attempt = 1) {
                 ui_report.hide_error($("#connection-error"));
             }
 
-            if (opts.msg_list.narrowed && opts.msg_list !== message_lists.current) {
+            // NOTE: opts.msg_list can be undefined here since load_messages is capable
+            // of handling such requests and hence we should retry the request instead of returning.
+            if (opts.msg_list !== message_lists.current && opts.msg_list.narrowed) {
                 // We unnarrowed before getting an error so don't
                 // bother trying again or doing further processing.
                 return;
```

Comment:
Isn't this still going to crash if `opts.msg_list` is `undefined` and `message_lists.current` is not?

Reply:
Yeah, fixed.